src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
       addresses = response.data.addresses;
     } else {
-      // TODO What error should be handled here?
+      throw new WalletRequestError('Error getting wallet addresses.');
     }
     return addresses;
   }
 
+  private async getNewAddresses() {
+    this.checkWalletReady();
+    const response = await walletApi.getNewAddresses(this);
+    let addresses: AddressInfoObject[] = [];
+    if (response.status === 200 && response.data.success === true) {
+      addresses = response.data.addresses;
+      this.newAddresses = addresses;
+      this.indexToUse = 0;
+    } else {
+      throw new WalletRequestError('Error getting wallet addresses to use.');
+    }
+  }
+
   /**
    * Get the balance of the wallet for a specific token
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async getBalance(token: string | null = null) {
-    const response = await walletApi.getBalances(this.walletId!, token);
-    let balance = null;
+  async getBalance(token: string | null = null): Promise<GetBalanceObject[]> {
+    this.checkWalletReady();
+    const response = await walletApi.getBalances(this, token);
+    let balance: GetBalanceObject[] = [];
     if (response.status === 200 && response.data.success === true) {
       balance = response.data.balances;
     } else {
-      // TODO What error should be handled here?
+      throw new WalletRequestError('Error getting wallet balance.');
     }
     return balance;
   }
 
   /**
    * Get the history of the wallet for a specific token
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async getTxHistory(options: { token?: string } = {}) {
+  async getTxHistory(options: { token?: string } = {}): Promise<GetHistoryObject[]> {
     // TODO Add pagination parameters
+    this.checkWalletReady();
     const requestOptions = Object.assign({ token: null }, options);
     const { token } = requestOptions;
-    const response = await walletApi.getHistory(this.walletId!, token);
-    let history = []
+    const response = await walletApi.getHistory(this, token);
+    let history: GetHistoryObject[] = []
     if (response.status === 200 && response.data.success === true) {
       history = response.data.history;
     } else {
-      // TODO What error should be handled here?
+      throw new WalletRequestError('Error getting wallet history.');
     }
     return history
   }
 
+  /**
+   * Get utxo from tx id and index
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  async getUtxoFromId(txId: string, index: number): Promise<Utxo | null> {
+    const response = await walletApi.getUtxos(this, { txId, index });
+    if (response.status === 200 && response.data.success === true) {
+      const utxos = response.data.utxos;
+      if (utxos.length === 0) {
+        // No utxo for this txId/index or is not from the requested wallet
+        return null;
+      } else {
+        return utxos[0];
```

Comment:
Why don't we return all utxos?

Reply:
This is a method that get a utxo from txId and index, so it must return one utxo, or null.

Reply:
Should we assert that `utxos.length == 1`?

Reply:
Done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/214/commits/2c8f734fab81b253460db4cf59e6742742b2d119